test/unit/lib/DiscoveryStreamFeed.test.js

```diff
@@ -440,6 +440,100 @@ describe("DiscoveryStreamFeed", () => {
+      ]);
+    });
+    it("should remove items with scores lower than min_score", () => {
+      const result = feed.transform({
+        spocs: [
+          {campaign_id: 2, item_score: 0.8, min_score: 0.9},
+          {campaign_id: 3, item_score: 0.7, min_score: 0.7},
+          {campaign_id: 1, item_score: 0.9, min_score: 0.8},
+        ],
+      });
+
+      assert.deepEqual(result.spocs, [
+        {campaign_id: 1, item_score: 0.9, score: 0.9, min_score: 0.8},
+        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.7},
+      ]);
+    });
+    it("should add a score prop to spocs", () => {
+      const result = feed.transform({
+        spocs: [
+          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
+        ],
+      });
+
+      assert.equal(result.spocs[0].score, 0.9);
+    });
+    it("should filter out duplicate campigns", () => {
+      const result = feed.transform({
+        spocs: [
+          {campaign_id: 2, item_score: 0.8, min_score: 0.1},
+          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
+          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
+          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
+          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
+        ],
+      });
+
+      assert.deepEqual(result.spocs, [
+        {campaign_id: 1, item_score: 0.9, score: 0.9, min_score: 0.1},
+        {campaign_id: 2, item_score: 0.8, score: 0.8, min_score: 0.1},
+        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.1},
+      ]);
+    });
+    it("should filter out duplicate campigns while using spocs_per_domain", () => {
+      sandbox.stub(feed.store, "getState").returns({
+        DiscoveryStream: {
+          spocs: {spocs_per_domain: 2},
+        },
+      });
+
+      const result = feed.transform({
+        spocs: [
+          {campaign_id: 2, item_score: 0.8, min_score: 0.1},
+          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
+          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
+          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
+          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
+          {campaign_id: 2, item_score: 0.8, min_score: 0.1},
+          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
+          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
+          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
+          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
+        ],
+      });
+
+      assert.deepEqual(result.spocs, [
+        {campaign_id: 1, item_score: 0.9, score: 0.9, min_score: 0.1},
+        {campaign_id: 1, item_score: 0.9, score: 0.9, min_score: 0.1},
+        {campaign_id: 2, item_score: 0.8, score: 0.8, min_score: 0.1},
+        {campaign_id: 2, item_score: 0.8, score: 0.8, min_score: 0.1},
+        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.1},
+        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.1},
+      ]);
+    });
+  });
+
```

Comment:
trying for this case with item_scores below is failing test, 

      const result = feed.transform({
        spocs: [
          {campaign_id: 2, item_score: 0.8, min_score: 0.1},
          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
          {campaign_id: 2, item_score: 0.9, min_score: 0.1},
          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
          {campaign_id: 1, item_score: 0.8, min_score: 0.1},
          {campaign_id: 3, item_score: 0.7, min_score: 0.1},
          {campaign_id: 1, item_score: 0.9, min_score: 0.1},
        ],
      });

      assert.deepEqual(result.spocs, [
        {campaign_id: 1, item_score: 0.9, score: 0.9, min_score: 0.1},
        {campaign_id: 1, item_score: 0.8, score: 0.9, min_score: 0.1},
        {campaign_id: 2, item_score: 0.9, score: 0.8, min_score: 0.1},
        {campaign_id: 2, item_score: 0.8, score: 0.8, min_score: 0.1},
        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.1},
        {campaign_id: 3, item_score: 0.7, score: 0.7, min_score: 0.1},
      ]);

Reply:
it was my miss, passing with above values, will good to have test updated with different item_scores to cover that